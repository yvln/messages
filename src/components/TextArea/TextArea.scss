@import "styles/_variables.scss";

.TextArea {
  display: flex;
  justify-content: center;
  margin-bottom: $spacing-small;

  > textarea {
    font-family: 'PT Sans', "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
    font-size: 15px;
    line-height: 1.3em;

    padding: 2%;
    width: 96%;
    height: 70px;

    border: 2px solid $color-lighter;
    border-radius: .3em;

    resize: none;
  }
}