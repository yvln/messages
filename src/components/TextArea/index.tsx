import * as React from 'react';

import './TextArea.scss';

export interface IProps {
  onChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
  value: string,
}

const TextArea = (props: IProps) => {
  return (
    <div className="TextArea">
      <textarea onChange={props.onChange} value={props.value} />
    </div>
  );
};

export default TextArea;
