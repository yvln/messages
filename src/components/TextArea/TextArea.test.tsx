import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import TextArea, { IProps } from '.';

configure({ adapter: new Adapter() });

describe('src/components/TextArea', () => {
  let props: IProps;

  beforeEach(() => {
    props = {
      onChange: jest.fn(),
      value: 'This is a value',
    };
  });

  it('should render', () => {
    const component = shallow(<TextArea {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should display its value', () => {
    const component = mount(<TextArea {...props} />);
    expect(component.text()).toEqual('This is a value');
  });
});
