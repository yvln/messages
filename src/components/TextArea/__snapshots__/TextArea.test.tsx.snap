// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/TextArea should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextArea
    onChange={[MockFunction]}
    value="This is a value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <textarea
        onChange={[MockFunction]}
        value="This is a value"
      />,
      "className": "TextArea",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": [MockFunction],
        "value": "This is a value",
      },
      "ref": null,
      "rendered": null,
      "type": "textarea",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <textarea
          onChange={[MockFunction]}
          value="This is a value"
        />,
        "className": "TextArea",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [MockFunction],
          "value": "This is a value",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
